### ISSUE TITLE
[ENHANCEMENT] Bump `rancher-monitoring` chart version

### ISSUE THREAD

**Issue by w13915984028** [harvester/harvester#3780]
**Is your enhancement related to a problem? Please describe.**

In PR https://github.com/harvester/harvester-installer/pull/402 we tried to bump `rancher-monitoring` chart version, but the dashboard is broken, it needs more investigation.

**Describe the solution you'd like**

Bump `rancher-monitoring` chart version in Harvester v1.2.0 release and solve the potential issues.

**Describe alternatives you've considered**

**Additional context**

The `rancher-monitoring managedchart` has already been addon in https://github.com/harvester/harvester/issues/2992

EPIC:
https://github.com/harvester/harvester/issues/3262

PR is added, https://github.com/harvester/harvester-installer/pull/492

note: there is an UI issue to solve, DO NOT MERGE before it is solved.

**Comment by guangbochen** [comment]
@w13915984028 any updates on this one? thanks.

**Comment by w13915984028** [comment]
@guangbochen Not yet. Planned to investigate in these days, but gave way to coornidator, will continue next week.

**Comment by w13915984028** [comment]
comparing to Harvester v1.1.2, after upgrade `rancher-monitoring` to `102.0.1+up40.1.2` in Harvester v1.2.0, a new error from `rancher-monitoring-operator-76476f6486-jl9gh` is found.

> level=error ts=2023-05-16T18:40:42.151369158Z caller=klog.go:116 component=k8s_client_runtime func=ErrorDepth msg="status \"cattle-monitoring-system/rancher-monitoring-prometheus\" failed: failed to update status subresource: prometheuses.monitoring.coreos.com \"rancher-monitoring-prometheus\" not found"

```
...
level=info ts=2023-05-16T18:40:40.097052363Z caller=operator.go:1390 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="sync prometheus"
level=info ts=2023-05-16T18:40:40.099566719Z caller=operator.go:753 component=alertmanageroperator key=cattle-monitoring-system/rancher-monitoring-alertmanager msg="sync alertmanager"
level=info ts=2023-05-16T18:40:42.122103431Z caller=operator.go:1559 component=prometheusoperator key=cattle-monitoring-system/rancher-monitoring-prometheus msg="update prometheus status"
level=error ts=2023-05-16T18:40:42.151369158Z caller=klog.go:116 component=k8s_client_runtime func=ErrorDepth msg="status \"cattle-monitoring-system/rancher-monitoring-prometheus\" failed: failed to update status subresource: prometheuses.monitoring.coreos.com \"rancher-monitoring-prometheus\" not found"
```

<img width="1038" alt="image" src="https://github.com/harvester/harvester/assets/31133476/8a3731b9-cc38-49cf-a9c0-6bcb9cb782e7">


tracked in new issue https://github.com/harvester/harvester/issues/3969

**Comment by w13915984028** [comment]
this error message still exists in v1.2.0 with upgrade rancher-monitoring to `102.0.1+up40.1.2`

```
kubectl logs -n cattle-monitoring-system prometheus-rancher-monitoring-prometheus-0
```
> ts=2023-05-16T18:56:18.111Z caller=manager.go:636 level=warn component="rule manager" file=/etc/prometheus/rules/prometheus-rancher-monitoring-prometheus-rulefiles-0/harvester-system-prometheus-kubevirt-rules-518db8a5-6647-4c78-abfd-7e90112716b0.yaml group=kubevirt.rules name=KubeVirtComponentExceedsRequestedMemory index=37 msg="Evaluating rule failed" rule="alert: KubeVirtComponentExceedsRequestedMemory\nexpr: ((kube_pod_container_resource_requests{container=~\"virt-controller|virt-api|virt-handler|virt-operator\",namespace=\"harvester-system\",resource=\"memory\"})\n  - on (pod) group_left (node) container_memory_working_set_bytes{container=\"\",namespace=\"harvester-system\"})\n  < 0\nfor: 5m\nlabels:\n  kubernetes_operator_component: kubevirt\n  kubernetes_operator_part_of: kubevirt\n  severity: warning\nannotations:\n  description: Container {{ $labels.container }} in pod {{ $labels.pod }} memory usage\n    exceeds the memory requested\n  runbook_url: https://kubevirt.io/monitoring/runbooks/KubeVirtComponentExceedsRequestedMemory\n  summary: The container is using more memory than what is defined in the containers\n    resource requests\n" err="found duplicate series for the match group {pod=\"kube-vip-cloud-provider-0\"} on the right hand-side of the operation: [{__name__=\"container_memory_working_set_bytes\", endpoint=\"https-metrics\", id=\"/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod0b7e5252_e8de_4efd_9a99_14414a5ee831.slice/cri-containerd-9d7978e1072948242e15a691a31de01e9e4fe9ce8cce1a2f69e6333537fcbe51.scope\", image=\"docker.io/rancher/pause:3.6\", instance=\"192.168.122.50:10250\", job=\"kubelet\", metrics_path=\"/metrics/cadvisor\", name=\"9d7978e1072948242e15a691a31de01e9e4fe9ce8cce1a2f69e6333537fcbe51\", namespace=\"harvester-system\", node=\"harv120\", pod=\"kube-vip-cloud-provider-0\", service=\"rancher-monitoring-kubelet\"}, {__name__=\"container_memory_working_set_bytes\", endpoint=\"https-metrics\", id=\"/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod0b7e5252_e8de_4efd_9a99_14414a5ee831.slice\", instance=\"192.168.122.50:10250\", job=\"kubelet\", metrics_path=\"/metrics/cadvisor\", namespace=\"harvester-system\", node=\"harv120\", pod=\"kube-vip-cloud-provider-0\", service=\"rancher-monitoring-kubelet\"}];many-to-many matching not allowed: matching labels must be unique on one side"

https://github.com/harvester/harvester/issues/3792

**Comment by w13915984028** [comment]
UI confirmed below metrics error issue is related to upstream https://github.com/rancher/dashboard/issues/8885

<img width="658" alt="image" src="https://github.com/harvester/harvester/assets/31133476/3bba4797-6b91-4fe7-b24c-36b840317bbc">

**Comment by w13915984028** [comment]
With a fix per https://github.com/harvester/harvester-installer/pull/492, we got  rancher-monitoring `102.0.0+up40.1.2` working.

![image](https://github.com/harvester/harvester/assets/31133476/7ca5c721-ee3e-45aa-b857-552edfc41412)

![image](https://github.com/harvester/harvester/assets/31133476/69522bda-1ab3-4f76-8c66-8917511a4219)

**Comment by irishgordo** [comment]
Moving this back to implement pending:
- https://github.com/harvester/harvester/issues/4087
- https://github.com/harvester/harvester-installer/pull/512

**Comment by noahgildersleeve** [comment]
Tested in master-173b0824-head. Verified that Grafana was showing up on both dashboard and in Grafana dashboard in Rancher with imported Harvester and Harvester web ui.

![Screenshot_20230622_195303](https://github.com/harvester/harvester/assets/83787952/f71b1c89-68cd-4448-baae-882d94da35ec)
![Screenshot_20230622_195257](https://github.com/harvester/harvester/assets/83787952/25326ff6-2ce5-44cf-bef6-770dd2bfd3b7)
![Screenshot_20230622_195245](https://github.com/harvester/harvester/assets/83787952/eea667d2-33b7-4afd-bf10-51ca2a157620)
![Screenshot_20230622_195239](https://github.com/harvester/harvester/assets/83787952/da77cfd6-eda2-4915-b98e-75506abdbe7a)
